src/mapping/star_align/script.py

```diff
@@ -0,0 +1,197 @@
+import re
+import tempfile
+import subprocess
+from pathlib import Path
+import tarfile
+import gzip
+import shutil
+
+## VIASH START
+par = {
+  # 'input': [
+  #   'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L001_R1_001.fastq.gz',
+  #   'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L001_R2_001.fastq.gz',
+  #   'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L002_R1_001.fastq.gz',
+  #   'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/tinygex_S1_L002_R2_001.fastq.gz'
+  # ],
+  'input': [ 'resources_test/cellranger_tiny_fastq/cellranger_tiny_fastq/' ],
+  'reference': 'resources_test/cellranger_tiny_fastq/cellranger_tiny_ref/star/',
+  'output': 'out'
+}
+meta = {
+  'config': 'src/mapping/star_align/.config.vsh.yaml',
+  'cpus': None,
+  'memory_b': None,
+  'memory_kb': None,
+  'memory_mb': None,
+  'memory_gb': None,
+  'memory_tb': None,
+  'memory_pb': None,
+  'temp_dir': '/tmp'
+}
+## VIASH END
+
+########################
+### Helper functions ###
+########################
+
+# fastqgz_regex = r'([A-Za-z0-9\-_\.]+)_S(\d+)_L(\d+)_([RI]\d+)_(\d+)\.fastq(\.gz)?'
+# regex for matching R[12] fastq(gz) files
+fastqgz_regex = r'([A-Za-z0-9\-_\.]+)_S(\d+)_L(\d+)_(R\d+)_(\d+)\.fastq(\.gz)?'
+
+# helper function for cheching whether something is a gzip
+def is_gz_file(path: Path) -> bool:
+  with open(path, 'rb') as file:
+    return file.read(2) == b'\x1f\x8b'
+
+# look for fastq files in a directory
+def search_fastqs(path: Path) -> list[Path]:
+  if path.is_dir():
+    print(f"Input '{path}' is a directory, traversing to see if we can detect any FASTQ files.", flush=True)
+    value_paths = [file for file in path.iterdir() if re.match(fastqgz_regex, file.name) ]
+    return value_paths
+  else:
+    return [path]
+
+# if {par_value} is a Path, extract it to a temp_dir_path and return the resulting path
+def extract_if_need_be(par_value: Path, par_name: str, temp_dir_path: Path) -> Path:
+  print(f'>> Check compression of --{par_name} with value: {par_value}', flush=True)
+
+  if par_value.is_file() and tarfile.is_tarfile(par_value):
+    # Remove two extensions (if they exist)
+    extaction_dir_name = Path(par_value.stem).stem
```

Comment:
Could make an `extract_tar` and `extract_gzip` function?

Reply:
True, although I think `extract_if_need_be` is also ok ^^